implementation/src/main/java/io/smallrye/config/ConfigValue.java

```diff
@@ -62,6 +107,7 @@ public static ConfigValueBuilder builder() {
         private String value;
         private String configSourceName;
         private int configSourceOrdinal;
+        private int lineNumber;
```

Comment:
This should probably default to -1 TBH.  If some sources are using this as an index, then 0 might be valid (think database rows or something like that).